streams/service/src/main/java/com/hortonworks/streamline/streams/service/SchemaResource.java

```diff
@@ -132,4 +133,61 @@ public Response getKafkaSourceSchema(@PathParam("topicName") String topicName,
         }
     }
 
+    @GET
+    @Path("/{schemaName}/versions")
+    @Timed
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getAllSchemaVersions(@PathParam("schemaName") String schemaName,
+                                         @Context SecurityContext securityContext) {
+        try {
+            LOG.info("Get all versions for schema : {}", schemaName);
+            Collection<SchemaVersionInfo> schemaVersionInfos = schemaRegistryClient.getAllVersions(schemaName);
+            Collection<String> schemaVersions = new ArrayList<>();
+            if (schemaVersionInfos != null && !schemaVersionInfos.isEmpty()) {
+                LOG.debug("######### Received schema versions from schema registry: {}", schemaVersionInfos);
+                for (SchemaVersionInfo schemaVersionInfo: schemaVersionInfos) {
+                    schemaVersions.add(schemaVersionInfo.getVersion().toString());
+                }
+                return WSUtils.respondEntities(schemaVersions, OK);
+            } else {
+                LOG.debug("######### Received null response for schema versions from schema registry for schema: {}", schemaName);
+                return WSUtils.respondEntity(schemaVersions, NOT_FOUND);
+            }
+        } catch (SchemaNotFoundException e) {
+            LOG.error("Schema not found: [{}]", schemaName, e);
+            throw EntityNotFoundException.byId(schemaName);
```

Comment:
This should be `EntityNotFoundException#byName` instead of `EntityNotFoundException#byId`